cardano-api/test/Test/Cardano/Api/MetaData.hs

```diff
@@ -44,43 +47,24 @@ genTxMetadataValue =
   -- This shinks towards the head of the list, so have TxMetaMap at the end.
   Gen.choice
     [ TxMetaNumber <$> Gen.integral (Range.linear 0 10000)
-    , TxMetaBytes <$> genByteString
     , TxMetaText <$> genText
-    -- Only generate list of flat values to avoid generating a valid HashMap.
-    , TxMetaList <$> Gen.list (Range.linear 1 8) genFlatTxMetadataValue
-    , TxMetaMap <$> genPairList (Range.linear 1 8)
+    , TxMetaList <$> Gen.list (Range.linear 1 2) genTxMetadataValue
+    , TxMetaMap . Map.toList <$> Gen.map (Range.linear 1 4) ((,) <$> genValidJSONKey <*> genTxMetadataValue)
     ]
 
--- As above, but without nested values (List and Map).
-genFlatTxMetadataValue :: Gen TxMetadataValue
-genFlatTxMetadataValue =
-  Gen.choice
-    [ TxMetaNumber <$> Gen.integral (Range.linear 0 10000)
-    , TxMetaBytes <$> genByteString
-    , TxMetaText <$> genText
-    ]
+genValidJSONKey :: Gen TxMetadataValue
+genValidJSONKey = TxMetaText <$> genText
 
-genPairList :: Range Int -> Gen [(TxMetadataValue, TxMetadataValue)]
-genPairList len = do
-  keys <- Gen.list len $
-            Gen.frequency
-              [ (3, TxMetaText <$> genText)
-              , (2, TxMetaBytes <$> genByteString)
-              , (1, genFlatTxMetadataValue)
-              ]
-  -- Keys need to be unique and sorted.
-  -- They need to be unque because they are inserted into a HashMap and they
-  -- need to be sorted so they round trip correctly.
-  mapM (\k -> (k,) <$> genTxMetadataValue) (List.sort $ List.nub keys)
+genText :: Gen Text
+genText = Gen.choice
+  [ Gen.ensure (not . Text.isPrefixOf bytesPrefix)
+      (Text.pack <$> Gen.list (Range.linear 0 64) Gen.alphaNum)
```

Comment:
This looks suspiciously like it is generating a subset of the real world so that is can pass round tripping.

Reply:
Fixed.